### ISSUE TITLE
min/maxDate: can't be set at init

### ISSUE THREAD

**Issue by Zanguu** [Eonasdan/bootstrap-datetimepicker#851]
I have 2 datepickers to select a period for a timetable.
So i set the `format` option as `'HH:mm'`.

My period couldn't start before the school opens and can't end after the school closes. So i want to add the options `{minDate:'8:00', maxDate:'18:00'}`.
### First try
```javascript
    options = {
        format: 'HH:mm', extraFormats: ['HH:mm:ss'],
        minDate:'8:00', maxDate:'18:00'
    };
    $(this).datetimepicker(options)
```
>    TypeError: minDate() Could not parse date parameter: 8:00 

### Second try
```javascript
    options = {
        format: 'HH:mm', extraFormats: ['HH:mm:ss'],
        minDate:moment({h:8, m:0}), maxDate:moment({'18:00'})
    };
    $(this).datetimepicker(options)
```
This worked. Seemed awkward to me but I got proof that it was usable for the project.
But I needed to set it configurable by `data-date-*` attributes.

### Last try
```html
    <div class="input-group time" data-date-min-date="moment{h:8, m:0}">
        <input name="startDate" />
        <span class="input-group-addon"><span class="glyphicon glyphicon-time"></span></span>
    </div>
```
```javascript
    options = { format: 'HH:mm', extraFormats: ['HH:mm:ss']};
    $(this).datetimepicker(options)
```
I get the same error as first try.
After some investigation, seems that when parsing the minDate and maxDate, the format is not loaded into the private object `parseFormats` used to parse the moment in `parseInputDate`.
```javascript
    date = moment(date, parseFormats, options.useStrict);
```
### Temporary Workaround
```html
    <div class="input-group time">
        <input name="startDate" data-min-date="8:00" />
        <span class="input-group-addon"><span class="glyphicon glyphicon-time"></span></span>
    </div>
```
```javascript
    options = { format: 'HH:mm', extraFormats: ['HH:mm:ss']};
    $(this).datetimepicker(options)
        .data('DateTimePicker')
        .options($('input', this).data());
```
_I feel dirty._

## TL;DR
`parseFormats` is filled after the options are parsed which leads to error when giving _minDate/maxDate_ at init.

**Comment by Eonasdan** [comment]
Please check the dev branch. Disabled hours and disabled time intervals are new options

**Comment by Zanguu** [comment]
I tried _disabledTimeIntervals_: 
```html
    <div class="input-group time">
        <input name="startDate" data-date-disabled-time-intervals="[[9,19]]" />
        <span class="input-group-addon"><span class="glyphicon glyphicon-time"></span></span>
    </div>
```
No error, no effect.

I tried _enabledHours_ and _disabledHours_:
```html
    <div class="input-group time">
        <input name="startDate" data-date-disabled-hours="[0,1,2,3,4,5,6,7,8,20,21,22,23]" />
        <span class="input-group-addon"><span class="glyphicon glyphicon-time"></span></span>
    </div>
```
```html
    <div class="input-group time">
        <input name="startDate" data-date-enabled-hours="[9,10,11,12,13,14,15,16,17,18,19]" />
        <span class="input-group-addon"><span class="glyphicon glyphicon-time"></span></span>
    </div>
```
Does the job but 2 bad points:
- it's heavy to use, you have to put each single autorized (or not) hour in the attribute. The notation of _disabledTimeIntervals_ seems more accurate to me;
- you loose granularity after hours. I want to allow user to put 19:00 but not 19:30 (stepping of 30) and I can't.

As previously stated, the main bug was that the formats options are loaded after the date constraints. Maybe change the load order of the option would be the solution, but I don't have a clue on how to do that.

Thank you anyway for the correction.
